### ISSUE TITLE
MaxListenersExceededWarning

### ISSUE THREAD

**Issue by sebdeckers** [npm/gauge#82]
When instantiating 10 or more gauges in the same process there are warnings from Nodejs core.

```js
var G = require('gauge')

new G, new G, new G, new G, new G, new G, new G, new G, new G // first nine are okay

new G // tenth: resize
// > (node:71997) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 resize listeners added. Use emitter.setMaxListeners() to increase limit

new G  // eleventh: exit
// > (node:71997) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 exit listeners added. Use emitter.setMaxListeners() to increase limit
```

As a workaround:
- Limit the number of concurrent instances to 9.
- Always call `gauge.disable()` (where `gauge` is the instance of `G`).
- Pass the constructor the option: `{cleanupOnExit: false}` (your app should deal with the cleanup in `SIGINT`).

But, it would be nice if the library could manage this overhead. I might get around to digging deeper and create a PR. For now just wanted to record this issue in case anyone else (or future me) runs into the same problem.

**Comment by iarna** [comment]
Having more than one gauge at a time _is_ pretty weird. What's your use case for that?

Currently the way it's written the expectation is that the onExit handler should exist as long as the gauge object does. As we have no way of knowing when a gauge object goes away, we have to hold on to it and it's going to be eternal.

I'm up for patching it so that it only keeps the onExit active as long as the gauge is _enabled_.  This'd have the advantage of allowing disabled gauges to be garbage collected. Besides, having more than one `gauge` object enabled at a time is almost certainly an error as they'll write over each other. (If you have need of multiple gauges updating at once you're should be using `gauge/plumbing` and doing the screen updates yourself.)

**Comment by sebdeckers** [comment]
@iarna Thanks for looking at this. :) Proposed patch sounds perfect.

It's not technically necessary to have them at the same time, which would be epilepsy-inducing, but merely instantiated in the same process over time. I encountered this in a CLI app that watches files and re-processes them, showing a new gauge each time. Upon the tenth file change it threw the above `exit` error. At no point was there more than one gauge visible, but their global event listeners were essentially leaking.

**Comment by iarna** [comment]
Ah, that makes sense. I've got a patch for this and am meditating on how I want to test it, so it should land soonish. =)

**Comment by iarna** [comment]
Fixed in gauge@2.7.3! =)
